components/CollectionCells/examples/CollectionCellsDefaultUICollectionView.m

```diff
@@ -0,0 +1,157 @@
+  self.collectionView.alwaysBounceVertical = YES;
+  
+  _inkTouchController = [[MDCInkTouchController alloc] initWithView:self.collectionView];
+  _inkTouchController.delegate = self;
+    
+  // Register cell classes
+  [self.collectionView registerClass:[MDCCollectionViewTextCell class]
+          forCellWithReuseIdentifier:kReusableIdentifierItem];
+    
+  _content = [NSMutableArray array];
+  for (int i=0; i<10; i++) {
+    [_content addObject:@[[NSString stringWithFormat:@"(Left) Two line text %d", i],
+                          [NSString stringWithFormat:@"(Left) here is the detail text %d", i]]];
+  }
+}
+
+#pragma mark <UICollectionViewDataSource>
+
+- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
+  return 2;
+}
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
+  return [_content count]/2;
+}
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+  MDCCollectionViewTextCell *cell =
+  [collectionView dequeueReusableCellWithReuseIdentifier:kReusableIdentifierItem
+                                            forIndexPath:indexPath];
+  cell.textLabel.text = _content[indexPath.item][0];
+  cell.detailTextLabel.text = _content[indexPath.item][1];
+  cell.backgroundColor = [UIColor whiteColor];
+  [(MDCShadowLayer *)cell.layer setElevation:2.f];
+  return cell;
+}
+
+- (CGFloat)collectionView:(UICollectionView *)collectionView
+    cellHeightAtIndexPath:(NSIndexPath *)indexPath {
+  return MDCCellDefaultTwoLineHeight;
+}
+
+#pragma mark - <UICollectionViewDelegateFlowLayout>
+
+- (CGSize)collectionView:(UICollectionView *)collectionView
+                  layout:(UICollectionViewLayout *)collectionViewLayout
+  sizeForItemAtIndexPath:(NSIndexPath *)indexPath {
+  return CGSizeMake(self.collectionView.frame.size.width-16, MDCCellDefaultTwoLineHeight);
+}
+
+- (UIEdgeInsets)collectionView:(UICollectionView *)collectionView
+                        layout:(UICollectionViewLayout *)collectionViewLayout
+        insetForSectionAtIndex:(NSInteger)section {
+  return UIEdgeInsetsMake(8, 8, 0, 8);
+}
+
+#pragma mark - CatalogByConvention
+
++ (NSArray *)catalogBreadcrumbs {
+  return @[ @"Collection Cells", @"Cell with UICollectionViewController" ];
+}
+
++ (BOOL)catalogIsPrimaryDemo {
+  return NO;
+}
+
+- (void)collectionView:(UICollectionView *)collectionView
+didHighlightItemAtIndexPath:(NSIndexPath *)indexPath {
+  UICollectionViewCell *cell = [collectionView cellForItemAtIndexPath:indexPath];
+  CGPoint location = [collectionView convertPoint:_inkTouchLocation toView:cell];
+  
+  MDCInkView *inkView;
+  if ([cell respondsToSelector:@selector(inkView)]) {
+    inkView = [cell performSelector:@selector(inkView)];
+  } else {
+    return;
+  }
+
+  [inkView startTouchBeganAnimationAtPoint:location completion:nil];
```

Comment:
I wonder what a more formal mechanism for associating in with a cell would look like?